app/interactors/ratings.js

```diff
@@ -0,0 +1,23 @@
+const { notFoundError } = require('../errors');
+const { weetNotFoundErrorMessage, userNotFoundErrorMessage } = require('../helpers/constants');
+const logger = require('../logger');
+const { findRateWeet, updateRateWeet, createRateWeet } = require('../services/ratings');
+const { getWeetById } = require('../services/weets');
+const { getUserById } = require('../services/users');
+
+exports.rateWeetsInteractor = async (params, body) => {
+  const weetId = params.id;
+  const { ratingUserId, score } = body;
+  const weet = await getWeetById(weetId);
+  if (!weet) throw notFoundError(weetNotFoundErrorMessage);
+  const { userId } = weet;
+  const user = await getUserById(userId);
+  if (!user) throw notFoundError(userNotFoundErrorMessage);
+  const newPoints = user.points + score;
+  const previousRatedWeet = await findRateWeet(weetId, ratingUserId);
+  if (!previousRatedWeet) return createRateWeet({ ratingUserId, weetId, score, userId, newPoints });
+  const ratingId = previousRatedWeet.id;
+  if (previousRatedWeet.score !== score) return updateRateWeet({ ratingId, score, userId, newPoints });
```

Comment:
What would happen if a user A rate a weet with -1 and then change it to +1? If I get this right, after changing the vote, the final points will be the same as if the user never rate.